### PR TITLE
sample_n() and sample_frac() can use n()

### PR DESCRIPTION
**Pull request by romainfrancois**
the two functions are now based on `slice`, which allows to tap into hybrid such as `n()`, `n_distinct()` etc ...

as a bonus, we no longer need the grouped_df version because `slice` already takes care of evaluating the expression within groups. 

```r
#' @export
sample_n.data.frame <- function(tbl, size, replace = FALSE,
                                weight = NULL, .env = NULL) {
  if (!is_null(.env)) {
    inform("`.env` is deprecated and no longer has any effect")
  }

  size <- enquo(size)
  weight <- enquo(weight)

  weight <- enquo(weight)
  slice(tbl, sample.int(n(), check_size(!!size, n(), replace = replace), replace = replace, prob = !!weight))
}


#' @export
sample_frac.data.frame <- function(tbl, size = 1, replace = FALSE,
                                   weight = NULL, .env = NULL) {
  if (!is_null(.env)) {
    inform("`.env` is deprecated and no longer has any effect")
  }

  size <- enquo(size)
  weight <- enquo(weight)

  slice(tbl, sample.int(n(), round(n() * check_frac(!!size, replace = replace)), replace = replace, prob = !!weight))
}
```

### CODE REVIEW

File: R/dataframe.R [hunk 1]
```diff
@@ -235,8 +235,11 @@ sample_n.data.frame <- function(tbl, size, replace = FALSE,
     inform("`.env` is deprecated and no longer has any effect")
   }
 
-  weight <- eval_tidy(enquo(weight), tbl)
-  sample_n_basic(tbl, size, FALSE, replace = replace, weight = weight)
+  size <- enquo(size)
+  weight <- enquo(weight)
+
+  weight <- enquo(weight)
```
Comment (krlmlr) [inline R/dataframe.R]: This looks redundant.

File: R/dataframe.R [hunk 2]
```diff
@@ -247,29 +250,12 @@ sample_frac.data.frame <- function(tbl, size = 1, replace = FALSE,
...
-  if (frac) {
-    check_frac(size, replace)
-    size <- round(size * n)
-  } else {
-    check_size(size, n, replace)
-  }
-
-  idx <- sample.int(n, size, replace = replace, prob = weight)
-  tbl[idx, , drop = FALSE]
+  slice(tbl, sample.int(n(), round(n() * check_frac(!!size, replace = replace)), replace = replace, prob = !!weight))
```
Comment (krlmlr) [inline R/dataframe.R]: Is `round()` the correct operation here?

File: R/dataframe.R (hunk 1 above)
Comment (romainfrancois) [reply R/dataframe.R]: it is. I probably forgot to push my last commit 😬

File: R/dataframe.R (hunk 2 above)
Comment (romainfrancois) [reply R/dataframe.R]: this is what the original code did

File: R/dataframe.R (hunk 1 above)
Comment (krlmlr) [reply R/dataframe.R]: ?

**Pull request merged**